### PR TITLE
Update to golang 1.14.1

### PR DESCRIPTION
**Pull request by tfussell**
There are quite a few significant changes [in 1.14](https://blog.golang.org/go1.14) including performance improvements, and I think it's best if we don't get too far behind on our versions. We use the latest release of `architect` in many places so we'll need to do some testing before actually merging this PR. I just wanted to get the conversation started about whether/when we should upgrade.

Build error should be fixed once https://github.com/giantswarm/retagger/pull/384 is merged.

### CODE REVIEW

**Comment by MarcelMue** [comment]
CI is not happy.

**Comment by tfussell** [comment]
This is building correctly now that the images have been retagged.

**Comment by sslavic** [comment]
FYI noticed recently a thread on golang-nuts, it seems go 1.14.0 is not usable on mac https://github.com/golang/go/issues/37478
Not sure if we have any devs with mac, they won't be able to update / align local builds with CI tooling. I'm on linux so not affected.

**Comment by TheoBrigitte** [comment]
> Not sure if we have any devs with mac, they won't be able to update / align local builds with CI tooling. I'm on linux so not affected.

All good here, running macOS and latest go since 10 days.
```
~ $ go version
go version go1.14 darwin/amd64
~ $ sw_vers
ProductName:    Mac OS X
ProductVersion: 10.14.1
BuildVersion:   18B75
```

**Comment by Shurikken** [comment]
@TheoBrigitte @sslavic 

hi

i confirm with old mac (for me 2009 sierra) go 1.14 don't work

**Comment by sslavic** [comment]
Go 1.14.1 got published few hours ago which has that mac issue fixed.

**Comment by Shurikken** [comment]
> Go 1.14.1 a été publié il y a quelques heures et a résolu ce problème de mac.

but i have same probleme with go 1.14.1 & hugo

**Comment by tfussell** [comment]
I can also confirm that 1.14.0 appears to work fine on my machine with macOS 1.14.6 (MBP 2018). There are some small improvements in 1.14.1, but I think I'll go ahead and merge this unless there are any complaints and we'll bump minors as needed.

**Comment by kopiczko** [comment]
What's the latest here? Should we update to 1.14.1 and merge?

**Comment by tfussell** [comment]
> What's the latest here? Should we update to 1.14.1 and merge?

Sorry, I forgot about this. Now that 1.14.1 is out I think it makes sense to update. Too tired to do it tonight, but I'll take care of it tomorrow unless someone else wants to do it.

File: Dockerfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ COPY --from=ct /etc/ct/lintconf.yaml /etc/ct/lintconf.yaml
 ENV GOPATH /go
 ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
 
-ARG HELM_VERSION=v2.14.3
+ARG HELM_VERSION=v2.16.3
```
Comment (rossf7) [inline Dockerfile:22]: <3

**Comment by tfussell** [comment]
Updating retagger again.

**Comment by tfussell** [comment]
Going with this.

**Pull request merged**